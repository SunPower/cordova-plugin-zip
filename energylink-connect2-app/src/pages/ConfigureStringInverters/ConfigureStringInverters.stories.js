import { storiesOf } from '@storybook/react'
import React from 'react'
import { Provider } from 'react-redux'

import ConfigureStringInverters from '.'

import { configureStore } from 'state/store'

storiesOf('ConfigureStringInverters', module).add('Header', () => {
  const { store } = configureStore({
    pvs: {
      settingMetadata: false,
      setMetadataStatus: 'success'
    }
  })

  return (
    <div className="full-min-height pl-10 pr-10">
      <Provider store={store}>
        <ConfigureStringInverters />
      </Provider>
    </div>
  )
})
