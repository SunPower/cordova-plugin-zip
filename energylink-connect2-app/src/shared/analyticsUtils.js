import {
  __,
  compose,
  curry,
  gt,
  isEmpty,
  join,
  lensIndex,
  map,
  not,
  over,
  props,
  propSatisfies,
  toLower,
  toUpper,
  when
} from 'ramda'

export const isNotEmpty = compose(not, isEmpty)

export const capitalizeWord = when(
  propSatisfies(gt(__, 0), 'length'),
  compose(join(''), over(lensIndex(0), toUpper), toLower)
)

const stringValueOrEmptyString = stringValue => stringValue || ''

const getStringProps = curry(compose(map(stringValueOrEmptyString), props))

export const getUserProfile = getStringProps([
  'uniqueId',
  'firstName',
  'lastName',
  'email',
  'dealerName',
  'recordType'
])
