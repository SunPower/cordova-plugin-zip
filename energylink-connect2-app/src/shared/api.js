import Swagger from 'swagger-client'
let apiParty, apiSite, apiFirmware, apiScanSN

const requestOptions = access_token => ({
  requestInterceptor: req => {
    if (access_token) req.headers['Authorization'] = `Bearer ${access_token}`
    return req
  }
})

export async function getApiFirmware() {
  if (!apiFirmware)
    apiFirmware = await Swagger(process.env.REACT_APP_SWAGGER_FIRMWARE)
  return apiFirmware
}
export async function getApiPVS() {
  return await Swagger(process.env.REACT_APP_PVS_ADDRESS)
}
export async function getApiDevice(access_token) {
  return await Swagger(
    process.env.REACT_APP_SWAGGER_DEVICE,
    requestOptions(access_token)
  )
}
export async function getApiParty(access_token) {
  apiParty = await Swagger(
    process.env.REACT_APP_SWAGGER_PARTY,
    requestOptions(access_token)
  )
  return apiParty
}
export async function getApiSearch(access_token) {
  return await Swagger(
    process.env.REACT_APP_SWAGGER_SEARCH,
    requestOptions(access_token)
  )
}
export async function getApiSite(access_token) {
  if (!apiSite)
    apiSite = await Swagger(
      process.env.REACT_APP_SWAGGER_SITE,
      requestOptions(access_token)
    )
  return apiSite
}
export async function getApiScanSN() {
  if (!apiScanSN)
    apiScanSN = await Swagger(process.env.REACT_APP_SCAN_SERIAL_NUMBERS)
  return apiScanSN
}
export async function getApiAuth(access_token) {
  return await Swagger(
    process.env.REACT_APP_SWAGGER_AUTH,
    requestOptions(access_token)
  )
}
export const storageSwaggerTag = 'Commissioning'
