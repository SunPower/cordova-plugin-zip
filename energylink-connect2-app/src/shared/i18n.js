import { useMemo } from 'react'
import { useSelector } from 'react-redux'

import * as languages from './locales'
import * as localizedComponents from './localizedComponents'

const getTranslate = translations => (key, ...params) => {
  let value = translations[key] || key
  if (params) {
    params.forEach((p, idx) => {
      value = value.replace(`{${idx}}`, p)
    })
  }
  return value
}

export const useI18n = () => {
  const language = useSelector(state => state.language)
  const { locale = 'en' } = language
  return useMemo(() => getTranslate(languages[locale]), [locale])
}

export const useI18nComponent = () => {
  const language = useSelector(state => state.language)
  const { locale = 'en' } = language || {}

  return useMemo(
    () => componentName => {
      const path = `${locale}_${componentName}`
      const Component = localizedComponents[path] || {}
      if (typeof Component !== 'function') {
        throw new Error(`Component not found: ${path}`)
      }
      return Component()
    },
    [locale]
  )
}
