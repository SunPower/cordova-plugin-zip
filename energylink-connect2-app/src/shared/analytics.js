import appVersion from '../macros/appVersion.macro'
import { capitalizeWord, getUserProfile } from 'shared/analyticsUtils'
import { MIXPANEL_EVENT_QUEUED } from 'state/actions/analytics'
import { getAppleDeviceFamily } from 'shared/appleDevicesTable'
import { prop } from 'ramda'

const getAppAndDeviceProperties = () => {
  const { device } = window
  const { model, version, platform, manufacturer } = device

  let deviceModel
  let deviceBrowserVersion

  if (platform === 'iOS') {
    deviceModel = getAppleDeviceFamily(model)
    deviceBrowserVersion = version
  } else {
    deviceModel = `${manufacturer}, ${model}`
  }

  return {
    'App Build': appVersion(),
    'App Flavor': process.env.REACT_APP_FLAVOR,
    'OS Version': version,
    ...(deviceBrowserVersion ? { $browser_version: deviceBrowserVersion } : {}),
    $device: deviceModel
  }
}

export const loggedIn = user => {
  const { mixpanel } = window
  const {
    uniqueId: userId,
    firstName,
    lastName,
    email,
    dealerName,
    recordType: dealerType
  } = getUserProfile(user)
  const appAndDeviceProperties = getAppAndDeviceProperties()

  mixpanel.register(appAndDeviceProperties)
  mixpanel.identify(userId)
  mixpanel.people.set(appAndDeviceProperties)
  mixpanel.people.set({
    $first_name: firstName,
    $last_name: lastName,
    $email: email,
    'Dealer Name': dealerName,
    'Dealer Type': capitalizeWord(dealerType)
  })
  mixpanel.track('Login', { Success: true })

  return MIXPANEL_EVENT_QUEUED('Login - Success')
}

export const loginFailed = () => {
  const { mixpanel } = window
  const appAndDeviceProperties = getAppAndDeviceProperties()

  mixpanel.register(appAndDeviceProperties)
  mixpanel.track('Login', { Success: false })

  return MIXPANEL_EVENT_QUEUED('Login - Failed')
}

export const scanPVS = ({ pvsSN, entryMethod }, event) => {
  const { mixpanel } = window

  mixpanel.register({ 'PVS SN': pvsSN })
  mixpanel.track('Scan PVS Tag', {
    'Entry Method': entryMethod
  })

  return MIXPANEL_EVENT_QUEUED(event)
}

export const siteNotFound = ({
  uniqueId: userId,
  firstName,
  lastName,
  email,
  dealerName,
  recordType: dealerType
}) => {
  const { mixpanel } = window
  const appAndDeviceProperties = getAppAndDeviceProperties()

  mixpanel.register(appAndDeviceProperties)
  mixpanel.identify(userId)
  mixpanel.people.set(appAndDeviceProperties)
  mixpanel.people.set({
    $first_name: firstName,
    $last_name: lastName,
    $email: email,
    'Dealer Name': dealerName,
    'Dealer Type': capitalizeWord(dealerType)
  })
  mixpanel.track('Find Site', { Found: false })

  return MIXPANEL_EVENT_QUEUED('Find Site - site not found')
}

export const siteFound = (
  {
    uniqueId: userId,
    firstName,
    lastName,
    email,
    dealerName,
    recordType: dealerType
  },
  { city, st_id, postalCode, siteKey, commissioned }
) => {
  const { mixpanel } = window
  const appAndDeviceProperties = getAppAndDeviceProperties()

  mixpanel.register(appAndDeviceProperties)
  mixpanel.identify(userId)
  mixpanel.people.set(appAndDeviceProperties)
  mixpanel.people.set({
    $first_name: firstName,
    $last_name: lastName,
    $email: email,
    'Dealer Name': dealerName,
    'Dealer Type': capitalizeWord(dealerType)
  })
  mixpanel.track('Find Site', {
    Found: true,
    ...{
      $city: city,
      State: st_id,
      'Zip Code': postalCode,
      'Site ID': siteKey,
      Commissioned: commissioned
    }
  })

  return MIXPANEL_EVENT_QUEUED('Find Site - site found')
}

export const saveConfiguration = config => {
  const { mixpanel } = window
  mixpanel.track('Configure', config)
  return MIXPANEL_EVENT_QUEUED('Configure - submit site config')
}

export const saveInventory = (inventory, hasEss) => {
  const { mixpanel } = window

  mixpanel.register({
    'Equinox Storage Site': hasEss
  })

  mixpanel.track('Inventory', {
    'AC Module Count': prop('AC_MODULES', inventory),
    'ESS System Size': hasEss ? prop('ESS', inventory) : 'None',
    'Equinox Storage Site': hasEss
  })
  return MIXPANEL_EVENT_QUEUED('Inventory')
}

export const deviceResume = pageResume => {
  const { mixpanel } = window
  mixpanel.register({
    Page: pageResume
  })
  return MIXPANEL_EVENT_QUEUED('User opens app - Register page resume')
}

/**
 *
 * @param {string}[interfaceType] - The type of connection the user has. either can be  PVS, Cell, or Customer WiFi
 * @return {Action<null, null>}
 */
export const internetConnection = (interfaceType = '') => {
  const { mixpanel } = window
  mixpanel.register({
    'Internet Connection': interfaceType
  })
  return MIXPANEL_EVENT_QUEUED('User opens app - Register page resume')
}

export const firmwareUpdateEvent = data => {
  const { mixpanel } = window
  mixpanel.track('Firmware Update', {
    'Current FW Version': prop('fromFWVersion', data),
    'Current BN Version': prop('fromBuildNumber', data),
    'New FW Version': prop('toFWVersion', data),
    'New BN Version': prop('toBuildNumber', data),
    Success: prop('success', data)
  })
  return MIXPANEL_EVENT_QUEUED('Firmware update - Update successful')
}

export const timeMixPanelEvent = name => {
  if (!name) throw new Error('The name parameter is required')
  const { mixpanel } = window
  mixpanel.time_event(name)
}

export const commissionSite = ({ duration, timeConfiguring }) => {
  const { mixpanel } = window
  mixpanel.track('Commission Site', {
    $duration: duration,
    'Time Configuring': timeConfiguring
  })
  return MIXPANEL_EVENT_QUEUED(
    'Commission Site - first successful site configuration'
  )
}
