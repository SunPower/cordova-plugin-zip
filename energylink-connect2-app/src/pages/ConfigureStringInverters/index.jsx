import { clone, findIndex, map, propEq, propOr } from 'ramda'
import React, { useEffect, useState } from 'react'
import { useDispatch, useSelector } from 'react-redux'
import { useHistory } from 'react-router-dom'

import StringInverter from 'components/StringInverter'
import SwipeableSheet from 'hocs/SwipeableSheet'
import { useStringInverterSelector } from 'pages/StringInverters/OtherDevicesList'
import paths from 'routes/paths'
import { useI18n } from 'shared/i18n'
import { getStringInverters, isUnknownSerialNumber } from 'shared/utils'
import { FETCH_MODELS_INIT, UPDATE_DEVICES_LIST } from 'state/actions/devices'
import { SAVE_STRING_INVERTERS } from 'state/actions/stringInverters'

const WaitingClaimingDevices = () => {
  const { settingMetadata, setMetadataStatus } = useSelector(state => state.pvs)
  const t = useI18n()
  const history = useHistory()
  const [open, setOpen] = useState(false)

  useEffect(() => {
    if (settingMetadata) setOpen(true)
  }, [history, settingMetadata])

  const clickHandler = () => {
    if (settingMetadata === false && setMetadataStatus === 'success')
      return history.push(paths.PROTECTED.RMA_DEVICES.path)
  }
  return (
    <SwipeableSheet open={open} onChange={clickHandler}>
      <h1 className="has-text-white is-size-4">{t('CLAIMING_DEVICES')}</h1>
      <button className="button primary is-primary" onClick={clickHandler}>
        {t('CONTINUE')}
      </button>
    </SwipeableSheet>
  )
}
const PendingDeviceConfiguration = ({ open, close }) => {
  const t = useI18n()
  return (
    <SwipeableSheet open={open} onChange={close}>
      <div className="tile is-flex flex-column">
        <span className="has-text-centered has-text-white mb-10 has-text-weight-bold">
          {t('ATTENTION')}
        </span>
        <span className="has-text-centered has-text-white mt-10 mb-20">
          {t('CONFIG_MISSING')}
        </span>
        <button className="button is-primary" onClick={close}>
          {t('CONTINUE')}
        </button>
      </div>
    </SwipeableSheet>
  )
}

const buildStringInverterObject = device => {
  return {
    moduleCount: device.COUNT,
    panelModel: device.panelModel,
    serialNumber: device.SERIAL,
    productModelName: device.MODEL,
    operation: 'BIND',
    type: 'INVERTER'
  }
}

const ConfigureStringInverters = () => {
  const dispatch = useDispatch()
  const history = useHistory()
  const t = useI18n()

  const { found } = useSelector(state => state.devices)
  const { groupedDevices } = useStringInverterSelector()
  const [pendingConfig, setOpenPendingConfig] = useState(false)

  const inverters = propOr([], 'Inverter', groupedDevices)
  const stringInverters = getStringInverters(inverters)
  const [devices, setDevices] = useState(stringInverters)

  const finished = !devices.find(
    ({ SERIAL = '', COUNT = 0, panelModel = '' }) =>
      isUnknownSerialNumber(SERIAL) || COUNT === 0 || panelModel === ''
  )

  useEffect(() => {
    dispatch(FETCH_MODELS_INIT())
  }, [dispatch])

  useEffect(() => {
    if (devices.length !== stringInverters.length) {
      setDevices(stringInverters)
    }
  }, [stringInverters, devices.length])

  const updateDevice = (index, device) => {
    const devicesCopy = clone(devices)
    devicesCopy[index] = device
    setDevices(devicesCopy)
  }

  const saveInverters = () => {
    if (finished) {
      const deviceListCopy = clone(found)
      devices.forEach(device => {
        const inverterToModifyIndex = findIndex(
          propEq('SERIAL', device.SERIAL)
        )(deviceListCopy)
        deviceListCopy[inverterToModifyIndex].COUNT = device.COUNT
        deviceListCopy[inverterToModifyIndex].SERIAL = device.SERIAL
        deviceListCopy[inverterToModifyIndex].panelModel = device.panelModel
      })
      dispatch(UPDATE_DEVICES_LIST(deviceListCopy))
      const refactoredStringInverters = map(buildStringInverterObject, devices)
      dispatch(SAVE_STRING_INVERTERS(refactoredStringInverters))
      history.push(paths.PROTECTED.RMA_DEVICES.path)
    }
    setOpenPendingConfig(true)
  }

  const goBack = () => {
    history.push(paths.PROTECTED.ADD_STRING_INVERTERS.path)
  }
  return (
    <div className="fill-parent is-flex tile is-vertical has-text-centered pl-10 pr-10 mb-40">
      <div className="header mb-20">
        <span
          className="sp-chevron-left has-text-primary is-size-4 go-back"
          onClick={goBack}
        />
        <span className="is-uppercase has-text-weight-bold  page-title">
          {t('STRING_INVERTERS')}
        </span>
      </div>
      <div>
        {devices.map((device, index) => (
          <StringInverter
            key={index}
            device={device}
            updateDevice={updateDevice}
            index={index}
          />
        ))}
      </div>
      <div className="is-flex">
        <button
          className="button is-primary is-outlined is-uppercase is-fullwidth mr-10"
          onClick={goBack}
        >
          {t('CANCEL')}
        </button>
        <button
          className="button is-primary is-fullwidth ml-10"
          onClick={saveInverters}
        >
          {t('SAVE')}
        </button>
      </div>
      <WaitingClaimingDevices />
      <PendingDeviceConfiguration
        open={pendingConfig}
        close={() => setOpenPendingConfig(false)}
      />
    </div>
  )
}

export default ConfigureStringInverters
