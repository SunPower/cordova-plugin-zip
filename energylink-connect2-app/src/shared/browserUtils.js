import { createExternalLinkHandler } from 'shared/routing'

export function openUrl(url) {
  window.SafariViewController.isAvailable(function(available) {
    if (available) {
      window.SafariViewController.show({
        url: url,
        hidden: false, // default false. You can use this to load cookies etc in the background (see issue #1 for details).
        animated: false, // default true, note that 'hide' will reuse this preference (the 'Done' button will always animate though)
        transition: 'curl', // (this only works in iOS 9.1/9.2 and lower) unless animated is false you can choose from: curl, flip, fade, slide (default)
        enterReaderModeIfAvailable: false, // default false
        tintColor: '#ffffff', // default is ios blue
        barColor: '#15202e', // on iOS 10+ you can change the background color as well
        controlTintColor: '#15202e' // on iOS 10+ you can override the default tintColor
      })
    } else {
      // potentially powered by InAppBrowser because that (currently) clobbers window.open
      createExternalLinkHandler(url)()
    }
  })
}

export function dismissBrowser() {
  window.SafariViewController.hide()
}
