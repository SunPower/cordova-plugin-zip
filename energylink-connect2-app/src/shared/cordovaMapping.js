import { fileExists } from 'shared/fileSystem'

export const getMd5FromFile = (filePath = '') =>
  new Promise((resolve, reject) => {
    fileExists(filePath).then(fileEntry => {
      if (!fileEntry) throw Error("The file doesn't exist")
      window.md5chksum.file(fileEntry, resolve, reject)
    })
  })

export const checkMD5 = async (filePath = '', expectedMd5 = '') => {
  const md5 = await getMd5FromFile(filePath)
  if (expectedMd5 === md5) return true
  throw new Error('MD5 are not the same')
}

export const cdvfileToNativePath = (filePath = '') =>
  new Promise((resolve, reject) => {
    fileExists(filePath)
      .then(fileEntry => {
        if (!fileEntry) reject("The file doesn't exists")
        resolve(fileEntry.toURL())
      })
      .catch(reject)
  })

export const getMD5FromFile = (filePath = '') => {
  return new Promise((resolve, reject) => {
    cdvfileToNativePath(filePath)
      .then(nativePath => {
        window.FileHash.sha256(
          nativePath,
          ({ result }) => resolve(result),
          reject
        )
      })
      .catch(reject)
  })
}
