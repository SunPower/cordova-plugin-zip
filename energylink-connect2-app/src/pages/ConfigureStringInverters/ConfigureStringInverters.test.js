import React from 'react'

import ConfigureStringInverters from '.'

import * as i18n from 'shared/i18n'

describe('ConfigureStringInverters component', () => {
  beforeEach(() => {
    jest
      .spyOn(i18n, 'useI18n')
      .mockImplementation(() => (key = '', ...params) =>
        `${key.toUpperCase()} ${params.join('_')}`.trim()
      )
  })

  test('render correctly', () => {
    const { component } = mountWithProvider(<ConfigureStringInverters />)({
      pvs: {
        settingMetadata: {},
        setMetadataStatus: ''
      },
      devices: {
        found: []
      }
    })
    expect(component).toMatchSnapshot()
  })
})
