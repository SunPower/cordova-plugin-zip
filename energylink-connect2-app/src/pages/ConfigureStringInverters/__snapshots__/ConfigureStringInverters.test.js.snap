// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConfigureStringInverters component render correctly 1`] = `
<BrowserRouter
  keyLength={0}
>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <Provider
      store={
        Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        }
      }
    >
      <ConfigureStringInverters>
        <div
          className="fill-parent is-flex tile is-vertical has-text-centered pl-10 pr-10 mb-40"
        >
          <div
            className="header mb-20"
          >
            <span
              className="sp-chevron-left has-text-primary is-size-4 go-back"
              onClick={[Function]}
            />
            <span
              className="is-uppercase has-text-weight-bold  page-title"
            >
              STRING_INVERTERS
            </span>
          </div>
          <div />
          <div
            className="is-flex"
          >
            <button
              className="button is-primary is-outlined is-uppercase is-fullwidth mr-10"
              onClick={[Function]}
            >
              CANCEL
            </button>
            <button
              className="button is-primary is-fullwidth ml-10"
              onClick={[Function]}
            >
              SAVE
            </button>
          </div>
          <WaitingClaimingDevices>
            <SwipeableSheet
              onChange={[Function]}
              open={true}
            >
              <SwipeableBottomSheet
                defaultOpen={false}
                fullScreen={false}
                marginTop={0}
                onChange={[Function]}
                open={true}
                overflowHeight={0}
                overlay={true}
                scrollTopAtClose={true}
                shadowTip={true}
                swipeableViewsProps={
                  Object {
                    "className": "swipeable",
                  }
                }
                topShadow={true}
              >
                <div
                  style={
                    Object {
                      "bottom": 0,
                      "height": 0,
                      "left": 0,
                      "position": "fixed",
                      "right": 0,
                    }
                  }
                >
                  <HeightUpdater
                    height={768}
                    onHeightChange={[Function]}
                  />
                  <div
                    onClick={[Function]}
                    style={
                      Object {
                        "backgroundColor": "black",
                        "height": 768,
                        "left": 0,
                        "opacity": 0.54,
                        "pointerEvents": "auto",
                        "position": "fixed",
                        "right": 0,
                        "top": 0,
                        "transition": "opacity 450ms",
                      }
                    }
                  />
                  <ReactSwipableView
                    animateHeight={false}
                    animateTransitions={true}
                    axis="y"
                    className="swipeable"
                    containerStyle={
                      Object {
                        "boxShadow": "rgba(0, 0, 0, 0.156863) 0px -6px 5px",
                        "boxSizing": "border-box",
                      }
                    }
                    disableLazyLoading={false}
                    disabled={false}
                    enableMouseEvents={true}
                    hysteresis={0.6}
                    ignoreNativeScroll={false}
                    index={1}
                    onChangeIndex={[Function]}
                    onTransitionEnd={[Function]}
                    resistance={false}
                    slideStyle={
                      Object {
                        "boxSizing": "border-box",
                        "marginBottom": -0,
                        "overflow": "visible",
                      }
                    }
                    springConfig={
                      Object {
                        "delay": "0s",
                        "duration": "0.35s",
                        "easeFunction": "cubic-bezier(0.15, 0.3, 0.25, 1)",
                      }
                    }
                    style={
                      Object {
                        "boxSizing": "border-box",
                        "overflowY": "initial",
                      }
                    }
                    threshold={5}
                  >
                    <div
                      className="swipeable"
                      onMouseDown={[Function]}
                      onMouseLeave={[Function]}
                      onMouseMove={[Function]}
                      onMouseUp={[Function]}
                      onScroll={[Function]}
                      onTouchEnd={[Function]}
                      onTouchStart={[Function]}
                      style={
                        Object {
                          "boxSizing": "border-box",
                          "overflowY": "initial",
                        }
                      }
                    >
                      <div
                        className="react-swipeable-view-container"
                        style={
                          Object {
                            "WebkitFlexDirection": "column",
                            "WebkitTransition": "-webkit-transform 0.35s cubic-bezier(0.15, 0.3, 0.25, 1) 0s",
                            "boxShadow": "rgba(0, 0, 0, 0.156863) 0px -6px 5px",
                            "boxSizing": "border-box",
                            "direction": "ltr",
                            "display": "flex",
                            "flexDirection": "column",
                            "height": null,
                            "transition": "transform 0.35s cubic-bezier(0.15, 0.3, 0.25, 1) 0s",
                            "willChange": "transform",
                          }
                        }
                      >
                        <div
                          aria-hidden={false}
                          data-swipeable="true"
                          key=".1"
                          style={
                            Object {
                              "WebkitFlexShrink": 0,
                              "boxSizing": "border-box",
                              "flexShrink": 0,
                              "marginBottom": -0,
                              "overflow": "visible",
                              "width": "100%",
                            }
                          }
                        >
                          <div
                            style={
                              Object {
                                "marginBottom": 0,
                              }
                            }
                          />
                        </div>
                      </div>
                    </div>
                  </ReactSwipableView>
                  <div
                    style={
                      Object {
                        "bottom": -60,
                        "boxShadow": "rgba(0, 0, 0, 0.7) 0px 0px 30px",
                        "height": 60,
                        "left": "-50%",
                        "position": "fixed",
                        "transform": "translateY(50px)",
                        "transition": "transform 450ms",
                        "width": "200%",
                      }
                    }
                  />
                </div>
              </SwipeableBottomSheet>
            </SwipeableSheet>
          </WaitingClaimingDevices>
          <PendingDeviceConfiguration
            close={[Function]}
            open={false}
          >
            <SwipeableSheet
              onChange={[Function]}
              open={false}
            >
              <SwipeableBottomSheet
                defaultOpen={false}
                fullScreen={false}
                marginTop={0}
                onChange={[Function]}
                open={false}
                overflowHeight={0}
                overlay={true}
                scrollTopAtClose={true}
                shadowTip={true}
                swipeableViewsProps={
                  Object {
                    "className": "swipeable",
                  }
                }
                topShadow={true}
              >
                <div
                  style={
                    Object {
                      "bottom": 0,
                      "height": 0,
                      "left": 0,
                      "position": "fixed",
                      "right": 0,
                    }
                  }
                >
                  <HeightUpdater
                    height={768}
                    onHeightChange={[Function]}
                  />
                  <div
                    onClick={[Function]}
                    style={
                      Object {
                        "backgroundColor": "black",
                        "height": 768,
                        "left": 0,
                        "opacity": 0,
                        "pointerEvents": "none",
                        "position": "fixed",
                        "right": 0,
                        "top": 0,
                        "transition": "opacity 450ms",
                      }
                    }
                  />
                  <ReactSwipableView
                    animateHeight={false}
                    animateTransitions={true}
                    axis="y"
                    className="swipeable"
                    containerStyle={
                      Object {
                        "boxSizing": "border-box",
                      }
                    }
                    disableLazyLoading={false}
                    disabled={false}
                    enableMouseEvents={true}
                    hysteresis={0.6}
                    ignoreNativeScroll={false}
                    index={0}
                    onChangeIndex={[Function]}
                    onTransitionEnd={[Function]}
                    resistance={false}
                    slideStyle={
                      Object {
                        "boxSizing": "border-box",
                        "marginBottom": -0,
                        "overflow": "visible",
                      }
                    }
                    springConfig={
                      Object {
                        "delay": "0s",
                        "duration": "0.35s",
                        "easeFunction": "cubic-bezier(0.15, 0.3, 0.25, 1)",
                      }
                    }
                    style={
                      Object {
                        "boxSizing": "border-box",
                        "overflowY": "initial",
                      }
                    }
                    threshold={5}
                  >
                    <div
                      className="swipeable"
                      onMouseDown={[Function]}
                      onMouseLeave={[Function]}
                      onMouseMove={[Function]}
                      onMouseUp={[Function]}
                      onScroll={[Function]}
                      onTouchEnd={[Function]}
                      onTouchStart={[Function]}
                      style={
                        Object {
                          "boxSizing": "border-box",
                          "overflowY": "initial",
                        }
                      }
                    >
                      <div
                        className="react-swipeable-view-container"
                        style={
                          Object {
                            "WebkitFlexDirection": "column",
                            "WebkitTransition": "all 0s ease 0s",
                            "boxSizing": "border-box",
                            "direction": "ltr",
                            "display": "flex",
                            "flexDirection": "column",
                            "height": null,
                            "transition": "all 0s ease 0s",
                            "willChange": "transform",
                          }
                        }
                      >
                        <div
                          aria-hidden={false}
                          data-swipeable="true"
                          key=".0"
                          style={
                            Object {
                              "WebkitFlexShrink": 0,
                              "boxSizing": "border-box",
                              "flexShrink": 0,
                              "marginBottom": -0,
                              "overflow": "visible",
                              "width": "100%",
                            }
                          }
                        >
                          <div
                            className="ReactSwipeableBottomSheet--closed"
                            style={
                              Object {
                                "backgroundColor": "white",
                                "height": "initial",
                                "maxHeight": 768,
                                "overflow": "hidden",
                              }
                            }
                          >
                            <div
                              className="tile is-flex flex-column"
                            >
                              <span
                                className="has-text-centered has-text-white mb-10 has-text-weight-bold"
                              >
                                ATTENTION
                              </span>
                              <span
                                className="has-text-centered has-text-white mt-10 mb-20"
                              >
                                CONFIG_MISSING
                              </span>
                              <button
                                className="button is-primary"
                                onClick={[Function]}
                              >
                                CONTINUE
                              </button>
                            </div>
                          </div>
                        </div>
                      </div>
                    </div>
                  </ReactSwipableView>
                </div>
              </SwipeableBottomSheet>
            </SwipeableSheet>
          </PendingDeviceConfiguration>
        </div>
      </ConfigureStringInverters>
    </Provider>
  </Router>
</BrowserRouter>
`;
